### PR TITLE
rancher-compose 0.2.6

### PR DESCRIPTION
**Pull request by alex88**
I've added rancher-compose formula

### CODE REVIEW

File: Library/Formula/rancher-compose.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  sha256 "eb52c47ac42b2ecf5cb33d6c78faac8f91b040b23801e42638c4d247db2e1590"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = "#{buildpath}:#{buildpath}/Godeps/_workspace"
+    mkdir_p "#{buildpath}/src/github.com/rancherio"
+    ln_s buildpath, "#{buildpath}/src/github.com/rancherio/rancher-compose"
+    system "go", "build", "-ldflags", "-w -X github.com/rancherio/rancher-compose/version.VERSION #{version}", "-o", "bin/rancher-compose"
+    bin.install "bin/rancher-compose"
```
Comment (bfontaine) [inline Library/Formula/rancher-compose.rb]: You can merge these lines: `system "go", "build", "-ldflags", "-w -X github.com/rancherio/rancher-compose/version.VERSION #{version}", "-o", "#{bin}/rancher-compose"`

File: Library/Formula/rancher-compose.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  test do
+    system "bin/rancher-compose", "help"
```
Comment (bfontaine) [inline Library/Formula/rancher-compose.rb]: Please add a test that does something more substantial than e.g. `help` or `--version`. Thanks!

File: Library/Formula/rancher-compose.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class RancherCompose < Formula
+  homepage "https://github.com/rancher/rancher-compose"
```
Comment (bfontaine) [inline Library/Formula/rancher-compose.rb]: Please add a short description using `desc "..."` above this line.

File: Library/Formula/rancher-compose.rb (hunk 2 above)
Comment (alex88) [inline Library/Formula/rancher-compose.rb]: Sorry I've copied it from another formula, the problem is, any other command does something on a remote server, like awscli https://github.com/Homebrew/homebrew/blob/6c9b6fe98f0a7723132fc1b69f793b8caccd1f5e/Library/Formula/awscli.rb#L119

Comment (bfontaine) [inline Library/Formula/rancher-compose.rb]: Ok.

**Comment by alex88** [comment]
Should be done, let me know if it's ok so I'll squash

**Comment by ibuildthecloud** [comment]
@alex88 I'll see if we can officially maintain this formula from the Rancher side.  Thanks!

**Comment by alex88** [comment]
@ibuildthecloud as an homebrew tap?

**Comment by ibuildthecloud** [comment]
@alex88 Sorry, I have no clue what I'm talking about.  I'm a 100% linux user, so I don't know homebrew.  I just know users want this, so if there is a way we can ensure that homebrew stays up to date as we release new versions, we would gladly help.  @vincent99 or @cjellick might know more from the Rancher side, they are Mac users.

**Comment by alex88** [comment]
@ibuildthecloud for the homebrew side, I think it's just a matter of making a PR with the url/sha change, you can use your own github repo if you don't want to PR but it takes an extra step for the user

**Comment by vincent99** [comment]
I use homebrew but don't know anything about maintaining a formula in it.  Is putting in a PR for every release the way it's typically done? (Seems like maybe it is since this is PR 43000+...)

Our concern is we theoretically have a Rancher release every week; not all of them will include changes to `rancher-compose`, but a lot of them will.

**Comment by bfontaine** [comment]
@alex88 @vincent99 There are two solutions here:

1. Get `rancher-compose` as an Homebrew core formula. Upside: users can install it as `brew install rancher-compose` and it *might* get more visibility (I’ve no data on that). Downside: you’ll have to make a PR for each new release or any change to the installation process. If the new release doesn’t break anything it’ll usually be merged a few hours after its submission.
2. Use a **tap** ([instructions here](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap)), which is essentially your own Homebrew repo. Upside: you have total control over it; you can update formulae as you want and as frequently as you want. It’ll work exactly the same way for your users: `brew update` will update their local copy of your tap, `brew upgrade` will upgrade the outdated formulae. Downside: users will need to type a few more keystrokes to install your formulae: `brew install yourusername/yourtap/theformula` (e.g. `rancher/tap/rancher-compose`). Homebrew needs this prefix to know where to find the formula. Once have done that, they can refer to any formula from your tap without this prefix if their name doesn’t conflict with a core one.

**Comment by mikemcquaid** [comment]
What @bfontaine said, yeh.

**Comment by alex88** [comment]
@vincent99 @ibuildthecloud ?

**Comment by vincent99** [comment]
Ok we're good with option 1/PR for each release then.  Is there anything else needed to get @alex88 's PR for the initial formula accepted?

**Comment by mikemcquaid** [comment]
The test needs fixed (possibly a sandbox error): http://bot.brew.sh/job/Homebrew%20Pull%20Requests/32140/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_rancher_compose/

**Comment by DomT4** [comment]
Don't think it's a sandbox error. There's a typo/misunderstanding in the test:

```
 system "bin/rancher-compose", "help"
```

Should be either:
```
 system bin/"rancher-compose", "help"
```
Or
```
 system "#{bin}/rancher-compose", "help"
```

**Comment by vincent99** [comment]
@alex88 are you still interested in getting this merged?  If not we'll put in our own PR

**Comment by alex88** [comment]
@vincent99 sorry I forgot to update it, I've just pushed the changes, if it works I'm going to rebase and squash then it can be merged

**Comment by alex88** [comment]
Damn travis build failed due github rate limiting

**Comment by dunn** [comment]
We're not relying on Travis for anything yet, so as long as Jenkins builds fine we're OK.

**Comment by ibuildthecloud** [comment]
Just a heads up 0.3.0 is the current release of rancher-compose.

**Comment by alex88** [comment]
@dunn @mikemcquaid should I open a new one or just update this?

**Comment by dunn** [comment]
For future reference, either is fine!

**Pull request closed without merging**